File: caesar.c
#include <stdio.h>
#include <stdlib.h>
#include <cs50.h>
#include <ctype.h>
#include <string.h>

int main(int argc, string argv[]) //get the result from the commandline argument
{    
    if (argc != 2)   //if the commandline argument is not just 2 words
    {
        printf("Usage: ./caesar key\n");
        return 1;
    }
    else
    {
        int key = atoi(argv[1]); //turn key into integer
        if (key != 0)
        {
            string plaintext = get_string("plaintext: ");//prompt the plaintext from the user
            printf("ciphertext: ");
            for (int i = 0, n = strlen(plaintext); i < n; i++)
            {
                if (isalpha(plaintext[i])) //see if the plaintext is alphabetic 
                {
                    if (isupper(plaintext[i])) //see if the character is uppercase letter
                    {
                        printf("%c", (((plaintext[i] - 65) + key) % 26) + 65); //shift plaincase character by key
                    }
                    else if (islower(plaintext[i])) //see if the character is lowecase letter
                    {
                        printf("%c", (((plaintext[i] - 97) + key) % 26) + 97); //shift plaincase character by key
                    }
                }             
                else
                {
                    printf("%c", plaintext[i]); // if the plaintext is not alphabetic leave is as it is
                }
            }
            printf("\n");
        }
        else //if the key is not integer
        {
            printf("Usage: ./caesar key\n"); 
        }
    } 
    return 0;
}

File: caesar.py
from sys import argv
from cs50 import get_string

if len(argv) != 2:
    print("Usage: python caesar.py k")
k = int(argv[1])
if k != 0:
    p = get_string("plaintext: ")
    print("ciphertext: ", end="")
    for c in p:
        if c.isalpha():
            if c.isupper():
                u = (((ord(c) - 65) + k) % 26) + 65
                print(chr(u), end="")
            else:
                l = (((ord(c) - 97) + k) % 26) + 97
                print(chr(l), end="")
        else:
            print(c, end="")
    print()

else:
    print("Usage: python caesar.py k")

